django-stubs/test/client.pyi

```diff
@@ -72,26 +72,95 @@ class _RequestFactory(Generic[_T]):
-    def trace(self, path: str, secure: bool = ..., **extra: Any) -> _T: ...
+    def get(
+        self,
+        path: str,
+        data: _GetDataType = ...,
+        secure: bool = ...,
+        *,
+        headers: dict[str, Any] | None = ...,
+        **extra: Any
+    ) -> _T: ...
+    def post(
+        self,
+        path: str,
+        data: Any = ...,
+        content_type: str = ...,
+        secure: bool = ...,
+        *,
+        headers: dict[str, Any] | None = ...,
+        **extra: Any
+    ) -> _T: ...
+    def head(
+        self, path: str, data: Any = ..., secure: bool = ..., *, headers: dict[str, Any] | None = ..., **extra: Any
+    ) -> _T: ...
+    def trace(self, path: str, secure: bool = ..., *, headers: dict[str, Any] | None = ..., **extra: Any) -> _T: ...
     def options(
-        self, path: str, data: dict[str, str] | str = ..., content_type: str = ..., secure: bool = ..., **extra: Any
+        self,
+        path: str,
+        data: dict[str, str] | str = ...,
+        content_type: str = ...,
+        secure: bool = ...,
+        *,
+        headers: dict[str, Any] | None = ...,
+        **extra: Any
+    ) -> _T: ...
+    def put(
+        self,
+        path: str,
+        data: Any = ...,
+        content_type: str = ...,
+        secure: bool = ...,
+        *,
+        headers: dict[str, Any] | None = ...,
+        **extra: Any
+    ) -> _T: ...
+    def patch(
+        self,
+        path: str,
+        data: Any = ...,
+        content_type: str = ...,
+        secure: bool = ...,
+        *,
+        headers: dict[str, Any] | None = ...,
+        **extra: Any
+    ) -> _T: ...
+    def delete(
+        self,
+        path: str,
+        data: Any = ...,
+        content_type: str = ...,
+        secure: bool = ...,
+        *,
+        headers: dict[str, Any] | None = ...,
+        **extra: Any
     ) -> _T: ...
-    def put(self, path: str, data: Any = ..., content_type: str = ..., secure: bool = ..., **extra: Any) -> _T: ...
-    def patch(self, path: str, data: Any = ..., content_type: str = ..., secure: bool = ..., **extra: Any) -> _T: ...
-    def delete(self, path: str, data: Any = ..., content_type: str = ..., secure: bool = ..., **extra: Any) -> _T: ...
     def generic(
-        self, method: str, path: str, data: Any = ..., content_type: str | None = ..., secure: bool = ..., **extra: Any
+        self,
+        method: str,
+        path: str,
+        data: Any = ...,
+        content_type: str | None = ...,
+        secure: bool = ...,
+        *,
+        headers: dict[str, Any] | None = ...,
+        **extra: Any
     ) -> _T: ...
 
 class RequestFactory(_RequestFactory[WSGIRequest]): ...
 
 class _AsyncRequestFactory(_RequestFactory[_T]):
     def request(self, **request: Any) -> _T: ...
     def generic(
-        self, method: str, path: str, data: Any = ..., content_type: str | None = ..., secure: bool = ..., **extra: Any
+        self,
```

Comment:
It’s most likely here because Django redefines it, and when the stubs are generated, it’s regenerated. 
It would probably make sense to remove both and rely on the types in `RequestFactory`.

Reply:
See https://github.com/django/django/blob/51fdea68415c580cd753990e3238ae2e0f015263/django/test/client.py#L610